### PR TITLE
use IntelDeflaterFactory from the GKL

### PR DESCRIPTION
**Pull request by akiezun**
fixes https://github.com/broadinstitute/gatk/issues/1903

also updated the system properties in build.gradle and gatk-launch

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/cmdline/CommandLineProgram.java [hunk 1]
```diff
@@ -128,6 +129,13 @@ public Object instanceMainPostParseArgs() {
             System.setProperty("java.io.tmpdir", f.getAbsolutePath()); // in loop so that last one takes effect
         }
 
+        //Set defaults (note: setting them here means they are not controllable by the user)
+        BlockCompressedOutputStream.setDefaultDeflaterFactory(new IntelDeflaterFactory());
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/cmdline/CommandLineProgram.java]: Add a command-line arg to `CommandLineProgram` that allows the user to opt-out of using the `IntelDeflater`

File: src/main/java/org/broadinstitute/hellbender/cmdline/CommandLineProgram.java [hunk 2]
```diff
@@ -128,6 +129,13 @@ public Object instanceMainPostParseArgs() {
...
+        //Note: IntelDeflaterFactory (incorrectly) only loads the library and sets the usingIntelDeflater if some deflaters are actually created
+        //That means that the reported Deflater will be JdkDeflater even when IntelDeflater will actually be used.
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/cmdline/CommandLineProgram.java]: File a bug against htsjdk to fix this

Comment (akiezun) [inline src/main/java/org/broadinstitute/hellbender/cmdline/CommandLineProgram.java]: i filed a gkl bug report https://github.com/Intel-HLS/GKL/issues/8

**Comment by akiezun** [comment]
i'll grab it

**Comment by akiezun** [comment]
note: This PR does not reinstate the IntelDeflater for Spark. It's now a bit tricky because you need to call `BlockCompressedOutputStream.setDefaultDeflaterFactory(new IntelDeflaterFactory());` on every executor or on every writer factory. Maybe a solution would be to use `IntelDeflaterFactory` directly in Hadoop-BAM. @droazen wdyt?

File: src/main/java/org/broadinstitute/hellbender/cmdline/CommandLineProgram.java (hunk 1 above)
Comment (akiezun) [inline src/main/java/org/broadinstitute/hellbender/cmdline/CommandLineProgram.java]: done

**Comment by gspowley** [comment]
GKL 0.1.1 release in Maven Central resolves https://github.com/Intel-HLS/GKL/issues/8 (not the Spark issue)

**Comment by akiezun** [comment]
thanks @gspowley - using 0.1.1 now and the issue is fixed

**Pull request merged**